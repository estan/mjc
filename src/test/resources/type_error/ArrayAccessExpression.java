class ArrayAccessExpressionTest {
    public static void main(String[] args) {
        int i;
        int[] ia;
        long l;
        long[] la;
        boolean b;
        A cA;
        B cB;

        i = ia[ia]; // WRONG_INDEX_TYPE
        i = ia[l];  // WRONG_INDEX_TYPE
        i = ia[la]; // WRONG_INDEX_TYPE
        i = ia[b];  // WRONG_INDEX_TYPE
        i = ia[cA]; // WRONG_INDEX_TYPE

        l = la[ia];  // WRONG_INDEX_TYPE
        l = la[l];   // WRONG_INDEX_TYPE
        l = la[la];  // WRONG_INDEX_TYPE
        l = la[b];   // WRONG_INDEX_TYPE
        l = la[cA];  // WRONG_INDEX_TYPE

        i = i[i];    // NOT_ARRAY_TYPE
        i = l[i];    // NOT_ARRAY_TYPE
        i = b[i];    // NOT_ARRAY_TYPE
        i = cA[i];   // NOT_ARRAY_TYPE
    }
}
class A {}
class B {}
