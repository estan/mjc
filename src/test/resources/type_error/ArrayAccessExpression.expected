WRONG_INDEX_TYPE
WRONG_INDEX_TYPE
WRONG_INDEX_TYPE
WRONG_INDEX_TYPE
WRONG_INDEX_TYPE
WRONG_INDEX_TYPE
WRONG_INDEX_TYPE
WRONG_INDEX_TYPE
WRONG_INDEX_TYPE
WRONG_INDEX_TYPE
NOT_ARRAY_TYPE
NOT_ARRAY_TYPE
NOT_ARRAY_TYPE
NOT_ARRAY_TYPE
